src/foam/nanos/crunch/MinMaxCapabilityData.js

```diff
@@ -34,8 +34,24 @@ foam.CLASS({
         return [];
       },
       javaFactory: 'return new String[0];',
+      adapt: function(_, n){
+        if ( foam.core.String.isInstance(n) ){
+          return [
+            n
+          ];
+        }
+
+        if ( foam.core.Int.isInstance(n) ){
+          return [
+            n.toString()
+          ];       
+        }
+
+        return n;
+      },
       postSet: function (o, n) {
         if ( ! foam.util.equals(o, n) ) {
+          this.selectedDataStable = null;
```

Comment:
Is this done to publish to the propertyChange topic? If yes I think it would be clearer to do this after setting it `= n`:
```javascript
this.propertyChange('selectedDataStable', this.selectedDataStable$);
```

Reply:
Done.